packages/zoe/src/contractFacet/zcfZygote.js

```diff
@@ -326,31 +243,65 @@ export const makeZCFZygote = (
     getInstance: () => getInstanceRecord().instance,
   });
 
+  const handleOfferHandle = provideKindHandle(zcfBaggage, 'handleOfferObj');
   // handleOfferObject gives Zoe the ability to notify ZCF when a new seat is
   // added in offer(). ZCF responds with the exitObj and offerResult.
-  /** @type {HandleOfferObj} */
-  const handleOfferObj = Far('handleOfferObj', {
-    handleOffer: (invitationHandle, zoeSeatAdmin, seatData) => {
-      const zcfSeat = makeZCFSeat(zoeSeatAdmin, seatData);
+  const makeHandleOfferObj = defineDurableKind(handleOfferHandle, () => ({}), {
+    handleOffer: (_context, invitationHandle, seatData) => {
+      const zcfSeat = makeZCFSeat(seatData);
+      // TODO: provide a details that's a better diagnostic for the
+      // ephemeral offerHandler that did not survive upgrade.
       const offerHandler = takeOfferHandler(invitationHandle);
-      const offerResultP = E(offerHandler)(zcfSeat, seatData.offerArgs).catch(
-        reason => {
-          if (reason === undefined) {
-            const newErr = new Error(
-              `If an offerHandler throws, it must provide a reason of type Error, but the reason was undefined. Please fix the contract code to specify a reason for throwing.`,
-            );
-            throw zcfSeat.fail(newErr);
-          }
-          throw zcfSeat.fail(reason);
-        },
-      );
+      const offerResultP =
```

Comment:
Add a TODO that we support non-function offer handlers because we do not yet have support for durable functions.

Reply:
Durable functions rather than having to express a durable offer handler as an object.